documents/model.go

```diff
@@ -18,6 +18,22 @@ import (
 	"github.com/ethereum/go-ethereum/common/hexutil"
 )
 
+const (
```

Comment:
As sometimes we create the tree manually, we have to provide a deterministic compact name to each of the fields. Some of them might have a correspondent field in the coredocument and some dont. This keeps a map of the literal name and the compact name.

Reply:
Not an immediate action item, but we have to make sure that we don't accidentally create two fields with the same property (one in invoice and one in CoreDoc for example). 

We could achieve this using reserved statements and putting field numbers into there for the fields of the other documents.

```
message Foo {
  reserved 2, 15, 9 to 11;
  reserved "foo", "bar";
}
```

Reply:
Yes, agree the initial definition of the protocol documents should have that defined upfront. In this particular case, for low level general fields, I would argue to have those fields defined in the coredoc proto message, even though they are calculated every time.